component/common/loki/wal/watcher_metrics.go

```diff
@@ -58,6 +59,15 @@ func NewWatcherMetrics(reg prometheus.Registerer) *WatcherMetrics {
 			},
 			[]string{"id"},
 		),
+		replaySegment: prometheus.NewGaugeVec(
```

Comment:
I have some additional metrics that would be handy but if you want to do them in another more focused PR thats fine. WAL size in bytes, and WAL oldest timestamp and current timestamp. Kind of like segment but its more actionable for alerts.

Reply:
Down for checking that in another PR, but since our WAL writer uses prometheus'es wlog implementation, we inherit all this metrics https://github.com/prometheus/prometheus/blob/3450572840881c870012ef80bff1877568698638/tsdb/wlog/wlog.go#L196-L205. I think we don't have timestamps, so that one might be interesting for alerting on WAL not being written or too far behind (like to measure lag from wal write to watcher read).